File: src/main/java/org/gzipper/java/application/hashing/MessageDigestAlgorithm.java
/*
 * Copyright (C) 2018 Matthias Fussenegger
 *
 * This program is free software: you can redistribute it and/or modify it under
 * the terms of the GNU General Public License as published by the Free Software
 * Foundation, either version 3 of the License, or (at your option) any later
 * version.
 *
 * This program is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU General Public License for more
 * details.
 *
 * You should have received a copy of the GNU General Public License along with
 * this program. If not, see <http://www.gnu.org/licenses/>.
 */
package org.gzipper.java.application.hashing;

/**
 * Enumeration for mapping available message digest algorithm names.
 *
 * @author Matthias Fussenegger
 */
public enum MessageDigestAlgorithm {

    MD5("MD5"),
    SHA_1("SHA-1"),
    SHA_256("SHA-256"),
    SHA_384("SHA-384"),
    SHA_512("SHA-512");

    private final String _algorithmName;

    private MessageDigestAlgorithm(String providerName) {
        _algorithmName = providerName;
    }

    public final String getAlgorithmName() {
        return _algorithmName;
    }
}

File: src/main/java/org/gzipper/java/application/hashing/MessageDigestAlgorithm.kt
/*
 * Copyright (C) 2022 Matthias Fussenegger
 *
 * This program is free software: you can redistribute it and/or modify it under
 * the terms of the GNU General Public License as published by the Free Software
 * Foundation, either version 3 of the License, or (at your option) any later
 * version.
 *
 * This program is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU General Public License for more
 * details.
 *
 * You should have received a copy of the GNU General Public License along with
 * this program. If not, see <http://www.gnu.org/licenses/>.
 */
package org.gzipper.java.application.hashing

/**
 * Enumeration for mapping available message digest algorithm names.
 *
 * @author Matthias Fussenegger
 */
enum class MessageDigestAlgorithm(val algorithmName: String) {
    MD5("MD5"), SHA_1("SHA-1"), SHA_256("SHA-256"), SHA_384("SHA-384"), SHA_512("SHA-512");

    override fun toString(): String {
        return algorithmName
    }
}